microcloud/service/lxd.go

```diff
@@ -400,6 +400,105 @@ func (s LXDService) GetUplinkInterfaces(ctx context.Context, bootstrap bool, pee
 	return candidates, nil
 }
 
+// CephDedicatedInterface represents a dedicated interface for Ceph.
+type CephDedicatedInterface struct {
+	Name      string
+	Type      string
+	Addresses []string
+}
+
+// GetCephInterfaces returns a map of peer name to slice of CephDedicatedInterface that may be used to setup
+// a dedicated Ceph network.
+func (s LXDService) GetCephInterfaces(ctx context.Context, bootstrap bool, peers []mdns.ServerInfo) (map[string][]CephDedicatedInterface, error) {
+	clients := map[string]lxd.InstanceServer{}
+	networks := map[string][]api.Network{}
+	if bootstrap {
+		client, err := s.Client(ctx, "")
+		if err != nil {
+			return nil, err
+		}
+
+		networks[s.Name()], err = client.GetNetworks()
+		if err != nil {
+			return nil, err
+		}
+
+		clients[s.Name()] = client
+	}
+
+	for _, info := range peers {
+		// Don't include a local interface unless we are bootstrapping, in which case we shouldn't use the remote client.
+		if info.Name == s.Name() {
+			continue
+		}
+
+		client, err := s.remoteClient(info.AuthSecret, info.Address, CloudPort)
+		if err != nil {
+			return nil, err
+		}
+
+		networks[info.Name], err = client.GetNetworks()
+		if err != nil {
+			return nil, err
+		}
+
+		clients[info.Name] = client
+	}
+
+	candidates := map[string][]CephDedicatedInterface{}
+	for peer, nets := range networks {
+		for _, network := range nets {
+			// Skip managed networks.
+			if network.Managed {
+				continue
+			}
+
+			// We only want physical ethernet or VLAN interfaces.
+			if !shared.ValueInSlice(network.Type, []string{"physical", "bridge", "bond", "vlan"}) {
+				continue
+			}
+
+			state, err := clients[peer].GetNetworkState(network.Name)
+			if err != nil {
+				continue
+			}
+
+			// Only full L2 devices.
```

Comment:
Are there any docs you can link to to see the requirements?

Reply:
There isn't any actually. I implied that we desire net devices that are not receiving broadcast frames but regular L2 net devices. I'm removing this comment. But this function is very similar to the above `GetUplinkInterfaces` function which is also filtering out net devices of type `broadcast`

Reply:
fixed